### PR TITLE
Added Coin98 wallet.

### PR DESCRIPTION
**Pull request by leonardgee**

### CODE REVIEW

File: packages/coin98/src/adapter.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  WalletAdapter,
+  WalletAdapterEvents,
+  WalletNotConnectedError,
+  WalletNotFoundError,
+  WalletPublicKeyError,
+  WalletSignatureError,
+  WalletNotInstalledError
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+import bs58 from 'bs58'
```
Comment (jordansexton) [inline packages/coin98/src/adapter.ts:14]: I see there's a peer/dev dependency on `@types/bs58`, but this suggests that there should be a dependency on both `bs58` and `@types/bs58`, and remove the peer and dev deps (we publish both compiled `lib` and uncompiled TS `src` dirs to npm)

File: packages/coin98/src/adapter.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  signTransaction: (transaction: Transaction) => Promise<Transaction>;
+  isConnected: () => boolean,
+  connect: ()=> Promise<string[]>,
+  disconnect: ()=> Promise<void>,
+  request: (params: RequestParams) => Promise<SignTransactionResponse>
+}
+
+interface Coin98Wallet{
+  sol?: Coin98SolProvider,
+  provider: unknown
```
Comment (jordansexton) [inline packages/coin98/src/adapter.ts:38]: `provider` isn't used anywhere here.

File: packages/coin98/src/adapter.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  private _wallet: Coin98SolProvider | null;
+  private _publicKey: PublicKey | null;
+
+  constructor(config: Coin98WalletAdapterConfig = {}) {
+      super();
+      this._connecting = false;
+      this._wallet = null;
+      this._publicKey = null;
+      this.signTransaction = this.signTransaction.bind(this);
+      this.signAllTransactions = this.signAllTransactions.bind(this);
```
Comment (jordansexton) [inline packages/coin98/src/adapter.ts:63]: It shouldn't be necessary to bind these methods -- `signAllTransactions` uses an arrow function to invoke `signTransaction`

File: packages/coin98/src/adapter.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+          this._connecting = true;
+
+          const wallet = window.coin98?.sol;
+          if (!wallet) throw new WalletNotFoundError();
+          if (!wallet.isCoin98) throw new WalletNotInstalledError();
+
+          let account: string;
+          try {
+              const [address] = await wallet.connect();
+              account = address
```
Comment (jordansexton) [inline packages/coin98/src/adapter.ts:99]:
```suggestion
          let account: string;
          try {
              ([account] = await wallet.connect());
```

File: packages/wallets/src/wallets.ts [hunk 5]
```diff
@@ -27,6 +29,13 @@ export interface Wallet {
 
 export const ICONS_URL = 'https://raw.githubusercontent.com/solana-labs/wallet-adapter/master/packages/wallets/icons';
 
+export const getBitpieWallet = (config?: Coin98WalletAdapterConfig): Wallet => ({
+    name: WalletName.Coin98,
+    url: 'https://coin98.com/',
```
Comment (jordansexton) [inline packages/wallets/src/wallets.ts:34]:
```suggestion
    url: 'https://coin98.com',
```

Just removing the trailing slash for consistency between the wallets.

**Review by jordansexton — COMMENTED** [review]
Looks good! Can you run `yarn fmt` from the package?

File: packages/coin98/src/adapter.ts (hunk 3 above)
Comment (leonardgee) [reply packages/coin98/src/adapter.ts]: removed.

File: packages/coin98/src/adapter.ts (hunk 2 above)
Comment (leonardgee) [reply packages/coin98/src/adapter.ts]: removed

File: packages/coin98/src/adapter.ts (hunk 1 above)
Comment (leonardgee) [reply packages/coin98/src/adapter.ts]: fixed.

**Comment by jordansexton** [comment]
Merged, thanks!

**Pull request merged**